viadot/flows/genesys_to_adls.py

```diff
@@ -0,0 +1,228 @@
+            local_file_path (str, optional): The local path from which to upload the file(s). Defaults to None.
+            adls_file_path (str, optional): The destination path at ADLS. Defaults to None.
+            overwrite_adls (bool, optional): Whether to overwrite files in the data lake. Defaults to True.
+            adls_sp_credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary with
+            ACCOUNT_NAME and Service Principal credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET). Defaults to None.
+            credentials(dict, optional): Credentials for the genesys api. Defaults to None.
+        """
+        self.flow_name = name
+        self.media_type_list = media_type_list
+        self.queueIds_list = queueIds_list
+        self.data_to_post = data_to_post_str
+        self.environment = environment
+        self.schedule_id = schedule_id
+        self.report_url = report_url
+        self.report_columns = report_columns
+        self.start_date = start_date
+        self.end_date = end_date
+        self.days_interval = days_interval
+        self.local_file_path = local_file_path
+        self.adls_file_path = adls_file_path
+        self.overwrite_adls = overwrite_adls
+        self.adls_sp_credentials_secret = adls_sp_credentials_secret
+        self.credentials = credentials
+
+        super().__init__(*args, name=self.flow_name, **kwargs)
+
+        self.gen_flow()
+
+    def gen_flow(self) -> Flow:
+
+        to_csv = GenesysToCSV(credentials=self.credentials)
+
+        file_names = to_csv.bind(
+            media_type_list=self.media_type_list,
+            queueIds_list=self.queueIds_list,
+            data_to_post_str=self.data_to_post,
+            start_date=self.start_date,
+            end_date=self.end_date,
+            days_interval=self.days_interval,
+            environment=self.environment,
+            flow=self,
+        )
+
+        uploader = adls_uploader(
+            file_names=file_names,
+            adls_file_path=self.adls_file_path,
+            adls_sp_credentials_secret=self.adls_sp_credentials_secret,
+            flow=self,
+        )
+
+        uploader.set_upstream(file_names, flow=self)
+
+
+class GenesysReportToADLS(Flow):
+    def __init__(
+        self,
+        name: str,
+        columns: List[str] = None,
+        vault_name: str = None,
+        local_file_path: str = None,
+        output_file_extension: str = ".csv",
+        sep: str = "\t",
+        adls_file_path: str = None,
+        if_exists: Literal["replace", "append", "delete"] = "replace",
+        overwrite_adls: bool = True,
+        adls_sp_credentials_secret: str = None,
+        credentials_secret: str = None,
+        schedule_id: str = None,
+        *args: List[any],
+        **kwargs: Dict[str, Any]
+    ):
+        """Flow for downloading data from genesys API to Azure Data Lake in csv format by default.
```

Comment:
Description seems really similar to  GenesysToADLS. Also you don't need to specify what is the default output file extension here.